adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,508 @@
+
+        t._2.foreach((x) => {
+
+          val qualifier: String = x._1
+          val value: Array[Byte] = x._2
+
+          val mykeyFamilyQualifier = new KeyFamilyQualifier(rowKey, family, Bytes.toBytes(qualifier))
+
+          data += Tuple2(mykeyFamilyQualifier, value)
+
+        })
+
+        data.iterator
+      },
+      stagingFolder, familyHBaseWriterOptions,
+      compactionExclude = false,
+      HConstants.DEFAULT_MAX_FILE_SIZE)
+    ("hadoop fs -chmod -R 777 " + stagingFolder) !
+    val conn = ConnectionFactory.createConnection(conf)
+    val load = new LoadIncrementalHFiles(conf)
+    load.doBulkLoad(new Path(stagingFolder), conn.getAdmin, conn.getTable(TableName.valueOf(hbaseTableName)), conn.getRegionLocator(TableName.valueOf(hbaseTableName)))
+
+  }
+
+  def deleteGenotypeSamplesFromHBase(sc: SparkContext,
+                                     hbaseTableName: String,
+                                     inputsampleIds: List[String] = null,
+                                     sampleListFile: String = null,
+                                     numPartitions: Int = 0): Unit = {
+
+    val scan = new Scan()
+    scan.setCaching(100)
+    scan.setMaxVersions(1)
+
+    val conf = HBaseConfiguration.create()
+    val hbaseContext = new HBaseContext(sc, conf)
+
+    var sampleIds: List[String] = inputsampleIds
+    if (sampleListFile != null) {
+      sampleIds = Source.fromFile(sampleListFile).getLines.toList
+    }
+
+    sampleIds.foreach(sampleId => {
+      scan.addColumn(Bytes.toBytes("g"), Bytes.toBytes(sampleId))
+    })
+
+    val resultHBaseRDD: RDD[(ImmutableBytesWritable, Result)] = hbaseContext.hbaseRDD(TableName.valueOf(hbaseTableName), scan)
+
+    hbaseContext.bulkDelete[(ImmutableBytesWritable, Result)](resultHBaseRDD,
+      TableName.valueOf(hbaseTableName),
+      putRecord => {
+        val currDelete = new Delete(putRecord._2.getRow)
+        sampleIds.foreach((sample) => {
+          currDelete.addColumns(Bytes.toBytes("g"), Bytes.toBytes(sample))
+        })
+        currDelete
+      },
+      4)
+
+  }
+
+  def loadGenotypesFromHBaseToGenotypeRDD(sc: SparkContext,
+                                          hbaseTableName: String,
+                                          sampleIds: List[String],
+                                          sequenceDictionaryId: String,
+                                          queryRegion: ReferenceRegion = null, //one-based
+                                          numPartitions: Int = 0): GenotypeRDD = {
+
+    val sequenceDictionary = loadSequenceDictionaryFromHBase(hbaseTableName + "_meta", sequenceDictionaryId)
+    val sampleMetadata = loadSampleMetadataFromHBase(hbaseTableName + "_meta", sampleIds)
+    val genotypes = loadRDDofGenotypeFromHBase(sc, hbaseTableName, sampleIds, queryRegion, numPartitions)
+    GenotypeRDD(genotypes, sequenceDictionary, sampleMetadata)
+  }
+
+  def loadGenotypesFromHBaseToVariantContextRDD(sc: SparkContext,
```

Comment:
I'm not sure we need/want this method.

The way I've been interpreting the current API design is that `VariantContext` is only an intermediate `o.b.a.models` class for converting to/from htsjdk `VariantContext`, not really something that we want users to focus on.